src/xenia/kernel/xam/xam_net.cc

```diff
@@ -1356,18 +1356,23 @@ struct host_set {
 
   void Load(const x_fd_set* guest_set) {
     assert_true(guest_set->fd_count < 64);
-    this->count = guest_set->fd_count;
-    for (uint32_t i = 0; i < this->count; ++i) {
+
+    count = 0;
+    for (uint32_t i = 0; i < guest_set->fd_count; ++i) {
       auto socket_handle = static_cast<X_HANDLE>(guest_set->fd_array[i]);
       if (socket_handle == -1) {
-        this->count = i;
+        count = i;
         break;
       }
       // Convert from Xenia -> native
       auto socket =
           kernel_state()->object_table()->LookupObject<XSocket>(socket_handle);
-      assert_not_null(socket);
-      this->sockets[i] = socket;
+      if (!socket) {
+        continue;
```

Comment:
To fix tomorrow: Based on windows.
If FD_SET contains entry about socket that is already closed it MUST return -1 from function and 0x2736 (WSAENOTSOCK) from WSAGetLastError